src/cache.rs

```diff
@@ -153,7 +148,9 @@ impl Cache {
                 builder = builder.proxy(proxy);
             }
         }
-        let client = builder.build().unwrap_or_else(|_| Client::new());
+        let client = builder
+            .build()
+            .context("Could not initialize HTTP client")?;
```

Comment:
This is now semantically different: Before, we would fallback to default config if the proxies didn't work, now we don't do that anymore. I am fine with the decision (users can still unset the vars), but I am unsure this is the right PR for that change

Reply:
Yeah, see #247. In case you prefer an explicit PR, I restored that branch and reopened the PR. I'll have to rebase this branch if we merge.